lib/facility.rb

```diff
@@ -1,14 +1,52 @@
 class Facility
   attr_reader :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
+  def initialize(name, address, phone, services)
     @name = name
     @address = address
     @phone = phone
-    @services = []
+    @services = services
   end
 
   def add_services(service)
     @services << service
   end
+
+  def register_vehicles(vehicle)
+    case vehicle.type
+    when :antique
+      vehicle.register(:antique, 25)
+    when :ev
+      vehicle.register(:ev, 200)
+    else
+      vehicle.register(:regular, 100)
+    end
+  end
+
+  def administer_written_test(registrant)
+    if registrant.has_permit? && registrant.age >= 16
```

Comment:
Good job checking these two conditions, we also wanted to check the condition that the Facillity had the service that would allow for the written test.